### DISCUSSION TITLE
Enable helmfile to list dependencies (DAG)

### DISCUSSION THREAD

**Discussion by strainovic** [helmfile/helmfile#1180] [Ideas]
Hi all,

I'd like to use the helmfile template command to render the manifest for ArgoCD. The issue that I have is that ArgoCD does not respect dependencies until you add a `sync-waves` annotation to generated manifests. To do that, I need to get a list of releases processed by `needs` with information about sequential or parallel execution.

Something like this:

```
0 - database
1 - authentication
1 - access-control
2 - content
3 - spa-ui
...
```

Index means the order of deployment. If two or more releases have the same index number, they can be deployed in parallel.

Once I have it, I can inject proper ArgoCD annotations and it will properly handle dependencies.

ref: https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/

If you have any other solution, let's discuss it.

Thanks!

**Comment by yxxhero** [comment]
this: https://github.com/helmfile/helmfile/pull/1035
 is it ok for you?

**Reply by yxxhero** [reply]
@strainovic

**Reply by strainovic** [reply]
Hi @yxxhero,

Yes, that is exactly what I need. What is left to do before this PR gets merged?

Thanks!

**Comment by strainovic** [comment]
Hi @yxxhero,

Can we proceed with this PR? It looks promising.

Thanks

**Comment by strainovic** [comment]
I'm thinking about --output json|yaml, since it is more useful than plain text output. @yxxhero what do you think?

**Comment by strainovic** [comment]
Thanks to @vlpav030, it is available starting with https://github.com/helmfile/helmfile/releases/tag/v0.164.0
